internal/service/kyma/deletion/usecases/set_skr_kyma_state_deleting.go

```diff
@@ -0,0 +1,63 @@
+package usecases
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+	"github.com/kyma-project/lifecycle-manager/internal/result"
+	"github.com/kyma-project/lifecycle-manager/internal/result/kyma/usecase"
+)
+
+type KymaStatusRepo interface {
+	SetStateDeleting(ctx context.Context, namespacedName types.NamespacedName) error
+	Get(ctx context.Context, namespacedName types.NamespacedName) (*v1beta2.KymaStatus, error)
+}
+
+type SkrAccessSecretRepo interface {
+	Exists(ctx context.Context, kymaName string) (bool, error)
+}
+
+type SetSkrKymaStateDeleting struct {
+	kymaStatusRepo      KymaStatusRepo
+	skrAccessSecretRepo SkrAccessSecretRepo
+}
+
+func NewSetSkrKymaStateDeleting(kymaStatusRepo KymaStatusRepo,
+	skrAccessSecretRepo SkrAccessSecretRepo,
+) *SetSkrKymaStateDeleting {
+	return &SetSkrKymaStateDeleting{
+		kymaStatusRepo:      kymaStatusRepo,
+		skrAccessSecretRepo: skrAccessSecretRepo,
+	}
+}
+
+func (u *SetSkrKymaStateDeleting) IsApplicable(ctx context.Context, kyma *v1beta2.Kyma) (bool, error) {
+	if kyma.DeletionTimestamp.IsZero() {
+		return false, nil
+	}
+
+	if exists, err := u.skrAccessSecretRepo.Exists(ctx, kyma.GetName()); !exists || err != nil {
+		return false, err
+	}
+
+	status, err := u.kymaStatusRepo.Get(ctx, kyma.GetNamespacedName())
+	if err != nil {
+		return false, err
+	}
+
+	return status.State != shared.StateDeleting, nil
+}
+
+func (u *SetSkrKymaStateDeleting) Execute(ctx context.Context, kyma *v1beta2.Kyma) result.Result {
+	return result.Result{
+		UseCase: u.Name(),
+		Err:     u.kymaStatusRepo.SetStateDeleting(ctx, kyma.GetNamespacedName()),
+	}
+}
+
+func (u *SetSkrKymaStateDeleting) Name() result.UseCase {
+	return usecase.SetSkrKymaStateDeleting
```

Comment:
May sound a bit redundant, but can we establish in these usecases that the usecase.* that is returned here is taken from a private const on top, similar to the other state setting usecase? So the mapping is immediately visible.
also no strong feelings on it, just for consistency and clarity.
So all functions point to the same constant

Reply:
Let's also quickly talk about this one. Not sure if I understand you right. If I want to do a `const useCase = usecase.SetSkrKymaStateDeleting` at the top of the file, I get a conflict since we are sharing the same package between usecases and `useCase` const is already defined. I could go for `const useCaseSetSkrKymaStateDeleting = usecase.SetSkrKymaStateDeleting`, but then I am not sure if I address your concern.